src/main/java/com/techsophy/tsf/services/gateway/repository/KeycloakRealmRepository.java

```diff
@@ -48,6 +49,10 @@ public Mono<ClientRegistration> findByRegistrationId(String registrationId)
         return tenants.getRegistrations().stream().filter(s -> s.equals(registrationId)).map(s ->
                 Mono.fromCallable(() ->{
                     String secret = service.fetchClientDetails(s,false);
+                    if(StringUtils.isBlank(secret))
```

Comment:
pls confirm if the service will return a blank secret in case of error?

Reply:
using try catch